stix_shifter_utils/stix_translation/src/json_to_stix/json_to_stix_translator.py

```diff
@@ -190,169 +82,201 @@ def gen_dict_extract(self, key, var):
-                        if 'references' in item:
-                            item['references'] = str(item['references']) +'_' + str(indx)
-
-        return ds_map
-
-    def _transform(self, object_map, observation, ds_map, ds_key, obj):
-
-        to_map = obj[ds_key]
-
-        if ds_key not in ds_map:
-            if self.options.get('unmapped_fallback'):
-                if ds_key not in self.ds_key_map:
-                    self.logger.info(
-                        'Unmapped fallback is enabled. Adding {} attribute to the custom object'.format(ds_key))
-                    cust_obj = {"key": "x-" + self.data_source.replace("_", "-") + "." + ds_key, "object":
-                                "cust_object"}
-                    if to_map is None or to_map == '':
-                        self.logger.debug("Removing invalid value '{}' for {}".format(to_map, ds_key))
-                        return
-                    self._handle_cybox_key_def(cust_obj["key"], observation, to_map, object_map, cust_obj["object"])
-            else:
-                self.logger.debug('{} is not found in map, skipping'.format(ds_key))
-            return
-
-        if isinstance(to_map, dict):
-            self.logger.debug('{} is complex; descending'.format(to_map))
-            # If the object is complex we must descend into the map on both sides
-            for key in to_map.keys():
-                self._transform(object_map, observation, ds_map[ds_key], key, to_map)
-            return
-
-        # if the datasource fields is a collection of json object than we need to unwrap it and create multiple objects
-        if isinstance(to_map, list):
-            self.logger.debug('{} is a list; unwrapping.'.format(to_map))
-            for item in to_map:
-                if isinstance(item, dict):
-                    new_ds_map = self._update_object_key(ds_map[ds_key], to_map.index(item))
-                    for field in item.keys():
-                        self._transform(object_map, observation, new_ds_map, field, item)
-        
-        generic_hash_key = ''
-
-        # get the stix keys that are mapped
-        ds_key_def_obj = ds_map[ds_key]
-        if isinstance(ds_key_def_obj, list):
-            ds_key_def_list = ds_key_def_obj
-        else:
-            # Use callback function to run module-specific logic to handle unknown filehash types
-            if self.callback:
-                try:
-                    generic_hash_key = self.callback(obj, ds_key, ds_key_def_obj['key'], self.options)
-                except(Exception):
-                    return
-
-            ds_key_def_list = [ds_key_def_obj]
 
 
-        for ds_key_def in ds_key_def_list:
-            if ds_key_def is None or 'key' not in ds_key_def:
-                self.logger.debug('{} is not valid (None, or missing key)'.format(ds_key_def))
-                continue
+    def _check_stix_value_pattern(self, observable_key, return_value):
```

Comment:
You may want to rename this to `check_stix_value_regex` or something like that. This existing naming could be confusing considering the project translates STIX Patterns, which has nothing to do with this method.

Reply:
I can use the `_valid_stix_value` as its ancestor. 
https://github.com/opencybersecurityalliance/stix-shifter/blob/develop/stix_shifter_utils/stix_translation/src/json_to_stix/json_to_stix_translator.py#L151
the meaning of the both functions is pretty my the same.